sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -540,17 +540,20 @@ class AdaptiveQueryExecSuite
   }
 
   test("Avoid plan change if cost is greater") {
+    val testData2 = spark.table("testData2")
+    val newTestData2 = testData2.withColumn("c", testData2("a"))
+    newTestData2.createTempView("newTestData2")
     val origPlan = sql("SELECT * FROM testData " +
-      "join testData2 t2 ON key = t2.a " +
-      "join testData2 t3 on t2.a = t3.a where t2.b = 1").queryExecution.executedPlan
+      "join newTestData2 t2 ON key = t2.a " +
+      "join testData2 t3 on t2.c = t3.a where t2.b = 1").queryExecution.executedPlan
 
     withSQLConf(
       SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "true",
-      SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "80") {
+      SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "100") {
       val (plan, adaptivePlan) = runAdaptiveAndVerifyResult(
         "SELECT * FROM testData " +
-          "join testData2 t2 ON key = t2.a " +
-          "join testData2 t3 on t2.a = t3.a where t2.b = 1")
+          "join newTestData2 t2 ON key = t2.a " +
+          "join testData2 t3 on t2.c = t3.a where t2.b = 1")
```

Comment:
With the `BroadcastHashJoinExec` change, shuffle is no longer introduced, thus cost is better, thus SMJ is changed to BHJ. So, I introduced a new column and reference that column as a join key to keep the existing behavior.